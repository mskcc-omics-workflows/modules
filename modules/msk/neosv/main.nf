process NEOSV {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/neoantigen-utils-base:1.3.0':
        'docker.io/mskcc/neoantigen-utils-base:1.3.0' }"

    input:
    tuple val(meta),  path(inputBedpe), val(hlaString)
    tuple path(gtf),  path(cdna)

    output:
    tuple val(meta),       path("*.net.in.txt"),               emit: mutOut
    tuple val(meta),       path("*.WT.net.in.txt"),            emit: wtOut
    path "versions.yml",                                       emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def NEOSV_VERSION = 1.1

    """

    echo ${hlaString} | tr ',' '\n' | sed 's/^[ \\t]*//;s/[ \t]*\$//' > hla.txt
    awk 'NF {print substr(\$0,1,5)"*"substr(\$0,6)}' hla.txt > temp_file && mv temp_file hla.txt

    neosv --sv-file test.sv.bedpe \\
    --out ./ \\
    --hla-file hla.txt \\
    --gtf-file ${gtf} \\
    --cdna-file ${cdna} \\
    --pyensembl-cache-dir ./ \\
    --prefix ${prefix}


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        NEOSV: v${NEOSV_VERSION}
    END_VERSIONS

    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def NEOSV_VERSION = 1.1
    """
    touch ${prefix}.WT.net.in.txt
    touch ${prefix}.net.in.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        NEOSV: v${NEOSV_VERSION}
    END_VERSIONS
    """
}
