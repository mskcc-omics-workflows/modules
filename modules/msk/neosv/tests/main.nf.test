nextflow_process {

    name "Test Process NEOSV"
    script "../main.nf"
    process "NEOSV"

    tag "modules"
    tag "modules_nfcore"
    tag "neosv"
    tag "modules_msk"

    test("neosv - fa") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(file(params.test_data_mskcc['neoantigen']['svbedpe']), checkIfExists: true),
                    "HLA-A24:02,HLA-A24:02,HLA-B39:01,HLA-B39:01,HLA-C07:01,HLA-C06:02"
                    ]

                input[1] = [
                    file(params.test_data_mskcc['neoantigen']['gtf'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['cdna'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.mutOut,
                    process.out.wtOut,
                    ).match()
                }
            )
        }

    }


    test("neosv - fa - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('MUT_sequence_fa'),
                    "HLA"
                    ]
                input[1] = [file('gtf'), file('cdna')]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.mutOut,
                    process.out.wtOut
                    ).match()
                }
            )
        }

    }

}
