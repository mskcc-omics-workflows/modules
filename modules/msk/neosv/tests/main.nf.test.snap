{
    "neosv - fa": {
        "content": [
            [
                "versions.yml:md5,1d203387ca616b0fa7b43ad9181a3375"
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.WT.net.in.txt:md5,dc236f5aa955bb6087c8e713d2c8d2f4",
                        "test.net.in.txt:md5,646a1666817d924b926c0b3bd6f58af3"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.WT.net.in.txt:md5,dc236f5aa955bb6087c8e713d2c8d2f4"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-27T11:59:59.477226567"
    },
    "neosv - fa - stub": {
        "content": [
            [
                "versions.yml:md5,b970b6b530af12cc2f05bba7747125ee"
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.WT.net.in.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.net.in.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.WT.net.in.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-27T12:00:07.449738079"
    }
}